config/test.exs

```diff
@@ -20,3 +20,6 @@ config :demo, DemoWeb.Endpoint,
 
 # Print only warnings and errors during test
 config :logger, level: :warn
+
+# Only in tests, remove the complexity from the password encryption algorithm
+config :bcrypt_elixir, :log_rounds, 1
```

Comment:
We should support all three comeonin adapters. We should default to bcrypt except on Windows which should use pkbdf2. Each adapter will require a different line here.

Reply:
Argon2 is recommended [here](https://github.com/riverrun/comeonin/wiki/Choosing-the-password-hashing-library). Any reason not to use it as the default?

Reply:
https://blog.benpri.me/blog/2019/01/13/why-you-shouldnt-be-using-bcrypt-and-scrypt/

Reply:
Yes, I am aware folks will have different opinions and takes on the password hashing, that's why I prefer to follow recommendations from OWASP and similar and, AFAIK, it is still bcrypt. If someone has any updated insights or data on that, I would love to hear. :)

Reply:
It seems that the recommendation is still `bcrypt`, except in some cases, as stated in [OWASP: Password Storage Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html).

Reply:
Why not leave `pbkdf2` as default, since it works on all 3 platforms.

Reply:
any thoughts on passwordless auth? it solves a lot of hassle in pass world like reset, change, forgot, keeps email as source of truth and solve password hashing issues

Reply:
> 
> 
> any thoughts on passwordless auth? it solves a lot of hassle in pass world like reset, change, forgot, keeps email as source of truth and solve password hashing issues

In my opinion, passwordless auth should be avoided most of the time. I guess it depends on your user base, but in my experience, many people use their work email to sign up for services, even on non work related sites. Unless you can verify their identity some other way during sign up, that means they will lose access to their accounts when they change jobs, since you can't trust email update requests from any other email address. This is especially important if your project falls under GDPR.

Reply:
> services

even in that cenario, its safer passwordless since the emails belongs to the company and that could be abused by bad actors, I manage and managed the email myself a couple of times its usual to help someone to recover stuff left in company emails they dont own anymore, passwords dont solve this issue,  specially for car rentals, and some other kind of stuff they get important purchase info by mail, the mistake is to use a email you dont own, and when you know is passwordless you think twice to use one you do not own.
I understand that a lot of work was put in this pr but if you think about future solutions of auth, passwordless is one of the most promising for a lot of reasons.

Reply:
Someone is welcome to implement `mix phx.gen.passwordless`. :) This can be a great starting point.

Reply:
any opinions on https://github.com/zanderxyz/veil?